### PR TITLE
feat: add script to automate releases

### PR DESCRIPTION
**Pull request by cjthompson**
## Description

Add a script to automate creating release branches and merging to main

## Notice

Before submitting a pull request, please make sure you have answered the following:

- [X] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?
- [X] Do all new and existing tests pass? Does the linter pass?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [X] :nail_care: New Feature (Breaking/Non-breaking Change)

## Testing

1. ./scripts/release.sh release v1.0.5
2. ./scripts/release.sh main v1.0.5

### CODE REVIEW

**Comment by 0xdef1cafe** [comment]
i'm ok with this. long term i'd prefer we use our preferred scripting language (node) for maintainability but this is readable and fine, also happy to iterate on this as we need.

**Review by 0xdef1cafe — APPROVED** [review]

File: scripts/release.sh [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  exit
+fi
+
+if [[ -n $(git status -s) ]]; then
+  echo "Cannot continue because your repository is dirty"
+  exit
+fi
+
+echo "Fetching latest repository state..."
+git fetch &> /dev/null
```
Comment (asamere) [inline scripts/release.sh:29]: do we want to hide the error output? what happens if auth fails?

File: scripts/release.sh [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      Yes ) break;;
+      No ) exit;;
+    esac
+  done
+
+  echo ""
+  echo "Checking out main branch..."
+  git checkout "origin/main" &> /dev/null
+  echo "Merging release branch into main..."
+  git merge --no-ff "origin/releases/$2" -m "Merge branch 'releases/$2'"
```
Comment (asamere) [inline scripts/release.sh:77]: what's the reason for disallowing a fast forward here?

File: scripts/release.sh [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    esac
+  done
+
+  echo ""
+  git tag -a -m "$2" "$2"
+  git push origin HEAD:main --tags
+
+  echo ""
+  echo "Deleting release branch..."
+  git push --delete origin "releases/$2" &> /dev/null
```
Comment (asamere) [inline scripts/release.sh:93]: feel strongly about deleting release branches? i'd keep them around, at least for a while and perhaps prune periodically.

File: scripts/release.sh (hunk 1 above)
Comment (cjthompson) [reply scripts/release.sh:29]: the script crashes

**Review by theoboldfrazier — APPROVED** [review]

**Review by asamere — APPROVED** [review]

**Pull request merged**